### PR TITLE
[SPARK-25436] Bump master branch version to 2.5.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
In the dev list, we can still discuss whether the next version is 2.5.0 or 3.0.0. Let us first bump the master branch version to `2.5.0-SNAPSHOT`.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @yhuai @rxin @cloud-fan

File: docs/_config.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ include:
 
 # These allow the documentation to be updated with newer releases
 # of Spark, Scala, and Mesos.
-SPARK_VERSION: 2.4.0-SNAPSHOT
-SPARK_VERSION_SHORT: 2.4.0
+SPARK_VERSION: 2.5.0-SNAPSHOT
+SPARK_VERSION_SHORT: 2.5.0-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (gatorsmile) [reply docs/_config.yml]: yes

File: project/MimaExcludes.scala [hunk 2]
```diff
@@ -1202,6 +1206,7 @@ object MimaExcludes {
   }
 
   def excludes(version: String) = version match {
+    case v if v.startsWith("2.5") => v25excludes
```
Comment (dongjoon-hyun) [inline project/MimaExcludes.scala]: Do we need to update [MimaBuild.scala](https://github.com/apache/spark/blob/master/project/MimaBuild.scala#L91) in this PR together?
```scala
val previousSparkVersion = "2.2.0"
```

Comment (gatorsmile) [reply project/MimaExcludes.scala]: Sure

Comment (gatorsmile) [reply project/MimaExcludes.scala]: This needs to be addressed in a separate PR.

Comment (dongjoon-hyun) [reply project/MimaExcludes.scala]: Oh, I got it, @gatorsmile . Thanks.

**Review by srowen — COMMENTED** [review]
Yes we definitely need to bump it; I kind of thought we were moving to 3.0? but doesn't hurt to make this change now.

**Comment by dongjoon-hyun** [comment]
+1, LGTM.

**Pull request closed without merging**